### DISCUSSION TITLE
The porosity difference between PNM and Porespy

### DISCUSSION THREAD

**Discussion by morendong** [PMEAL/porespy#890] [Q&A]
Hello, everyone,
I used blobs in Porespy to generate images with one porosity,
and extracted the pore network through op.io.network_from_porosity, and set the geometry by geometry.spheres_and_cylinders, finally calculated the porosity by pore_volumn and throat_vlumn.
Compared to the two values of porosity, the porosity of PNM is less than that of Porespy obviously.
How to make two values to be close.
Thanks in advance.

**Comment by jgostick** [comment]
Hi @morendong, I know you use porespy and openpnm a lot, so I'm sorry if you already know about thse examples, but as a first pass, have you looked at:

[this](https://openpnm.org/examples/applications/adjusting_pore_size_distributions.html)
[this](https://openpnm.org/examples/applications/porosity.html)
[and this](https://openpnm.org/examples/applications/mercury_intrusion.html)

They explain lots of ways you can adjust the porosity of the pore network.

**Comment by morendong** [comment]
Thanks for your reply.
https://openpnm.org/examples/applications/porosity.html
I do as the example, the porosity is consistent, while I adjust the pore and throat as spheres_and_cylinders, which show "'spheres_and_cylinders' can only be applied when throat diameter is smaller than that of adjacent pores."

**Reply by jgostick** [reply]
The spheres_and_cylinders model has a mathematical infinite if the spheres are too close.  You can use the cones_and_cylinders model, or the hybrid_spheres_and_cylinders.
